charts/tileserver/templates/deployment.yaml

```
@@ -0,0 +1,93 @@
+---
+
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: {{ include "tileserver.fullname" . }}
+  labels:
+    {{- include "tileserver.labels" . | nindent 4 }}
+    {{- if $.Values.api.labels }}
+    {{- toYaml $.Values.api.labels | nindent 4 }}
+    {{- end }}
+
+  {{- if $.Values.api.annotations }}
+  annotations:
+    {{- toYaml $.Values.api.annotations | nindent 4 }}
+  {{- end }}
+
+spec:
+  replicas: {{ .Values.api.replicaCount }}
+  revisionHistoryLimit: {{ .Values.api.revisionHistory }}
+
+  selector:
+    matchLabels:
+      {{- include "tileserver.selectorLabels" . | nindent 6 }}
+
+  {{- with .Values.api.strategy }}
+  strategy:
+    {{- toYaml . | nindent 4 }}
+  {{- end }}
+
+  template:
+    metadata:
+      labels:
+        {{- include "tileserver.selectorLabels" . | nindent 8 }}
+
+    spec:
+      terminationGracePeriodSeconds: {{ .Values.api.terminationGracePeriodSeconds }}
+
+      volumes:
+        - configMap:
+            name: {{ include "tileserver.fullname" . }}
+          name: config-volume
+
+      {{- with .Values.api.nodeSelector }}
+      nodeSelector:
+        {{- toYaml . | nindent 8 }}
+      {{- end }}
+
+      {{- with .Values.api.affinity }}
+      affinity:
+        {{- toYaml . | nindent 8 }}
+      {{- end }}
+
```

Стоит добавить еще `tolerations` на случай если у них ноды огорожены taint'ами, и `imagePullSecrets` если registry с авторизацией.
```suggestion

          {{- with .Values.api.tolerations }}
          tolerations:
            {{- toYaml . | nindent 8 }}
          {{- end }}
          
          {{- with .Values.api.imagePullSecrets }}
          imagePullSecrets:
            {{- toYaml . | nindent 8 }}
          {{- end }}
```

---

добавил